apps/rush/src/RushVersionSelector.ts

```diff
@@ -27,27 +30,45 @@ export class RushVersionSelector {
       { node: process.versions.node }
     );
 
+    let installPromise: Promise<void> = Promise.resolve();
+
     if (!installMarker.isValid()) {
-      // Need to install Rush
-      console.log(`Rush version ${version} is not currently installed. Installing...`);
-
-      Utilities.installPackageInDirectory(
-        expectedRushPath,
-        isLegacyRushVersion ? '@microsoft/rush' : '@microsoft/rush-lib',
-        version,
-        'rush-local-install',
-        MAX_INSTALL_ATTEMPTS,
-        true
-      );
-
-      console.log(`Successfully installed Rush version ${version} in ${expectedRushPath}`);
-    }
+      installPromise = installPromise.then(() => {
+        // Need to install Rush
+        console.log(`Rush version ${version} is not currently installed. Installing...`);
+
+        const resourceName: string = `rush-${version}`;
+
+        console.log(`Trying to acquire lock for ${resourceName}`);
 
```

Comment:
This seems like the kind of thing that should only need to be printed in a "verbose" mode.

Reply:
We do this because `acquire()` can stall for several minutes while it waits for the other process to perform the installation.  Without some kind of message, the tool looks like it's frozen or stuck in a loop or something.

I thought about changing the messages to "Waiting for access to the X folder", but sometimes there is no wait at all.

---
In reply to: [185168800](https://github.com/Microsoft/web-build-tools/pull/632#discussion_r185168800) [](ancestors = 185168800)